### PR TITLE
.widget()

### LINKED ISSUE #8: Handle OSError due to in-use web server port automatically for the user
If the user does not specify a port when initializing `VitessceWidget()` *and* if the default port 8000 is already allocated, then try subsequent ports until an available one has been found.

This should allow the average user to ignore the port argument.

### LINKED ISSUE #13: Should more functions be object oriented?
Rather than `my_widget = VitessceWidget(config)` do we want the API to be `my_widget = config.widget()`?

And likewise for uploading to AWS S3? Rather than `upload_to_s3(config, s3, bucket_name)`, the API could be `config.upload_to_s3(s3, bucket_name)`

### LINKED ISSUE #16: Use Vitessce light theme by default
Alternatively, check whether the current environment is Jupyter and check if using the Jupyter dark theme. Or check for the operating system theme like here https://github.com/higlass/higlass/pull/837/files

### LINKED ISSUE #19: Annotate example notebooks
Explain each step or line of code in the example notebooks.

Update the README at `notebooks/README.md`

### PR DESCRIPTION
**Pull request by keller-mark**
Fixes #8
Fixes #13 
Fixes #16
Fixes #19

This PR adds a function `.widget()` to the `VitessceConfig` class to allow creating a widget without importing the additional `VitessceWidget` class. I have removed some old notebooks and have added more annotations to the other notebooks. This PR also finishes the half-implemented `VitessceConfig.from_dict` static method, which has a corresponding demo notebook.

### CODE REVIEW

File: vitessce/wrappers.py [hunk 1]
```diff
@@ -328,23 +329,31 @@ def create_cells_json(self):
...
-            cell_set = [
-                str(cell_id)
-                for cell_id, cell_cluster_id in cell_cluster_tuples
-                if cell_cluster_id == cluster_id
-            ]
-            cell_sets.add_node(str(cluster_id), ['Clusters'], cell_set)
+        if cell_set_obs_cols is not None:
 
-        return cell_sets.json
+            for clusters_key in cell_set_obs_cols:
```
Comment (ilan-gold) [inline vitessce/wrappers.py:339]: Maybe rename this variable from `clusters_key` to `tree_name` or something - it took me a second to figure out what was going on.

**Review by ilan-gold — CHANGES_REQUESTED** [review]
Got an error on the `pbmc` tutorial (it never loaded) but otherwise looked/worked ok:

```
Error while validating cell-sets. index.min.js:2
cell-sets from http://localhost:8003/A/0/cell-sets: validation failed [
  {
    "keyword": "type",
    "dataPath": "",
    "schemaPath": "#/type",
    "params": {
      "type": "object"
    },
    "message": "should be object"
  }
]
```

File: vitessce/wrappers.py (hunk 1 above)
Comment (keller-mark) [reply vitessce/wrappers.py]: Thanks, I have changed the name and added a comment

**Review by ilan-gold — CHANGES_REQUESTED** [review]
Can you get rid of the port/height binding on the higlass example?

**Comment by ilan-gold** [comment]
@keller-mark The higlass example also does not seem to work for me - does it work for you?  Nothing appears.
Screen Shot 2020-12-17 at 3 24 20 PM

**Comment by keller-mark** [comment]
I just updated the HiGlass notebook to remove the higlass view config and the port binding code. This can be tested by building the Vitessce bundle `npm run build-lib:prod` on this branch https://github.com/hubmapconsortium/vitessce/pull/803, copying it to `vitessce-python/js/node_modules/vitessce/dist` and running `npm run build` in `vitessce-python/js` and then `jupyter labextension install ../js` in `vitessce-python/notebooks`

**Comment by keller-mark** [comment]
Alternatively we can just wait to merge https://github.com/hubmapconsortium/vitessce/pull/803 and update the version in package.json here

**Comment by keller-mark** [comment]
I moved the notebooks to `docs/notebooks` and added the `nbsphinx` package which renders them into the documentation site

**Review by ilan-gold — APPROVED** [review]
All notebooks except imaging work, which I am not so worried about.

**Pull request merged**